browser/brave_profile_prefs.cc

```diff
@@ -176,6 +176,12 @@ void RegisterProfilePrefsForMigration(
 
   // Added 01/2022
   registry->RegisterBooleanPref(brave_rewards::prefs::kHideButton, false);
+
+#if BUILDFLAG(ENABLE_EXTENSIONS)
+  // Added 11/2022
+  registry->RegisterBooleanPref(kDontAskEnableWebDiscovery, false);
+  registry->RegisterIntegerPref(kBraveSearchVisitCount, 0);
```

Comment:
Q. I'm kind of confused about this because we're clear these two in `chromium_src/chrome/browser/prefs/browser_prefs.cc`. Could you tell me why we need both?

Reply:
This is `registering`. To clear existing prefs at `browser_prefs.cc` they should be regisrterd first.

Reply:
Humm, yeah. I thought registering a pref to be cleared soon is kind of inefficient so wondered why we aren't doing something like

```
if (prefs_->HasPref(kDeprecatedPref))
  prefs_->ClearPref(kDprecatedPref);
```

Pairing registering and clearing seems to be preferred by folks, I guess.

Reply:
Sorry I can't catch about `Pairing registering and clearing`.
Registering is done via `PrefRegistrySyncable` and clearing is done via `PrefService`.
`RegisterProfilePrefsForMigration()` does registering all deprecated prefs and `MigrateObsoleteProfilePrefs()` clears all deprecated prefs' existing value. And we use different methods for registering currently used prefs

Reply:
Yeah I think we're on the same page.